drivers/entropy/entropy_esp32.c

```diff
@@ -58,8 +64,28 @@ static int entropy_esp32_get_entropy(const struct device *dev, uint8_t *buf,
 
 static int entropy_esp32_init(const struct device *dev)
 {
+	int ret = 0;
+
+#if defined(SOC_RNG_CLOCK_IS_INDEPENDENT)
+	const struct device *clock_dev =
+		DEVICE_DT_GET(DT_CLOCKS_CTLR(DT_NODELABEL(trng0)));
+	clock_control_subsys_t clock_subsys =
+		(clock_control_subsys_t)DT_CLOCKS_CELL(DT_NODELABEL(trng0), offset);
+
+	if (!device_is_ready(clock_dev)) {
+		return -ENODEV;
+	}
+
+	ret = clock_control_on(clock_dev, clock_subsys);
+
+	if (ret != 0) {
+		LOG_ERR("Error enabling TRNG clock");
```

Comment:
Could be useful to log the return value as well?

Reply:
At the moment we only have two states (success or -EALREADY) implemented in the driver, so I'm not sure it would add much info for now.